### PR TITLE
TF-596: Save screen size from web settings

### PR DESCRIPTION
**Pull request by cartmouse**
- [x] Code Review
- [x] Developer Testing
- [x] QA Review
- [x] Test Plan updated
- [x] PO Review (Optional)

https://user-images.githubusercontent.com/39350320/182135034-28a60430-22f1-452c-8e0d-c2a440238dcb.mp4

### CODE REVIEW

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx [hunk 1]
```diff
@@ -22,8 +23,9 @@ import { TIMEOUT_S } from './CalibrationScreen';
 interface CalibrationInstructionsProps {
     progress: number;
     containerStyle: CSSProperties;
+    top?: boolean;
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: We could make this isTop (and then pass in false for the bottom one) meaning we can remove the containerStyle props as this is just a different div style depending on top or bottom

Or could make CalibrationInstructionsTop and CalibrationInstructionsBottom which just calls CalibrationInstructions with the correct isTop boolean??

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx [hunk 2]
```diff
@@ -264,3 +281,36 @@ const progressStyle = (progress: number, isHovered: boolean): CSSProperties => {
...
+    const { isFullScreen, isZoomed } = useWindowSize();
+    const [isReady, setIsReady] = useState(false);
+
+    useEffect(() => {
+        const timeout = setTimeout(() => setIsReady(true), 200);
+
+        return () => clearTimeout(timeout);
+    }, []);
+
+    if (isZoomed && isReady) {
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: Could move the isReady to a check with encompasses the isZoom and !isFullScreen check

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx [hunk 3]
```diff
@@ -93,6 +102,17 @@ const CalibrationScreen: React.FC<CalibrationScreenProps> = ({ isHandPresent, on
         };
     }, []);
 
+    useEffect(() => {
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx]: Could this be in CalibrationManager instead?

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx [hunk 4]
```diff
@@ -124,30 +144,43 @@ const CalibrationScreen: React.FC<CalibrationScreenProps> = ({ isHandPresent, on
...
-                {displayHandIndicator ? (
-                    <CalibrationHandLostMessage timeToPosSelect={timeToPosSelect} />
-                ) : (
-                    <div style={{ height: handNotFoundHeight }} />
-                )}
-                <CalibrationTutorialVideo videoStyle={{ paddingTop: '3vh' }} />
-                <CalibrationCancelButton onCancel={onCancel} buttonStyle={{ marginTop: '30.5vh' }} />
+                <FullScreenPrompt promptStyle={{ position: 'fixed', top: '5vh' }} />
+                <CalibrationProgressCircle progress={progress} style={{ position: 'fixed', top: '15.5vh' }} />
+                <img
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx]: Had this thought before but didn't get round to it - could move the touch circle image inside CalibrationProgressCircle?

File: TF_Settings_Web/src/Styles/Camera/Camera.scss [hunk 5]
```diff
@@ -15,3 +15,31 @@
     margin-bottom: 40px;
     background-color: #363636;
 }
+
+.quick-setup-container {
+    position: relative;
+    width: 63.75%;
+}
+
+.quick-setup-overlay {
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Styles/Camera/Camera.scss]: Is this class no longer use?

File: TF_Settings_Web/src/Styles/Camera/Camera.scss [hunk 6]
```diff
@@ -15,3 +15,31 @@
     margin-bottom: 40px;
     background-color: #363636;
 }
+
+.quick-setup-container {
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Styles/Camera/Camera.scss]: Is this class no longer used

File: TF_Settings_Web/src/Styles/_variables.scss [hunk 7]
```diff
@@ -6,6 +6,7 @@ $ultraleap-green: #00eb85;
 $light-grey-gradient: linear-gradient(180deg, #5c5c5c 0%, #454545 100%);
 $dark-grey-gradient: linear-gradient(180deg, #373737 0%, #2c2c2c 100%);
 $base-font: MontserratRegular, Arial;
+$button-border-radius: 10px;
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Styles/_variables.scss]: This isn't used anywhere (expect the class above)

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx [hunk 8]
```diff
@@ -113,14 +129,15 @@ const ReturnToPositionScreenMessage: React.FC<HandsLostProps> = ({ timeToPosSele
     );
 };
 
-export const CalibrationHandLostMessage: React.FC<HandsLostProps> = ({ timeToPosSelect }) => {
+export const CalibrationHandLostMessage: React.FC<HandsLostProps> = ({ display, timeToPosSelect, handsLostStyle }) => {
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx:143]: Should the CalibrationHandLostMessage in CalibrationCompleteScreen be updated to use these new props?

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx (hunk 1 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: Updated name, left the rest as discussed.

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx (hunk 3 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx]: Moved :+1:

File: TF_Settings_Web/src/Styles/Camera/Camera.scss (hunk 5 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Styles/Camera/Camera.scss]: Removed

File: TF_Settings_Web/src/Styles/Camera/Camera.scss (hunk 6 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Styles/Camera/Camera.scss]: Removed

File: TF_Settings_Web/src/Styles/_variables.scss (hunk 7 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Styles/_variables.scss]: Removed

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx (hunk 4 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationScreen.tsx]: Done

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx (hunk 8 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx:143]: It now does :+1:

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx (hunk 2 above)
Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: Added

File: TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx [hunk 9]
```diff
@@ -79,11 +98,19 @@ export const CalibrationProgressCircle: React.FC<CalibrationProgressCircleProps>
...
+            className="touch-circle"
+            style={{ position: 'absolute', top: '22%', left: '22%' }}
+            src={FingerprintIcon}
+            alt="Fingerprint Icon showing where to place finger for Quick Setup"
+        />
     </div>
 );
 
 interface HandsLostProps {
+    display?: boolean;
```
Comment (JordanBerni-UL) [inline TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: Don't think this needs to be optional now (don't know if that matters)

Comment (cartmouse) [reply TF_Settings_Web/src/Components/Pages/Camera/QuickSetup/Calibration/CalibrationComponents.tsx]: Updated :+1:

**Review by JordanBerni-UL — APPROVED** [review]

**Pull request merged**